Update Postgres on Kubernetes tutorial (PD-355, PD-378, and PD-222) (#751)
portworx/pxdocs: content/portworx-install-with-kubernetes/application-install-with-kubernetes/postgresql/_index.md

---
title: PostgreSQL on Portworx
linkTitle: PostgreSQL
keywords: PostgreSQL, Postgres, install, kubernetes, k8s
description: Use this guide to install and run PostgreSQL using Kubernetes
weight: 3
noicon: true
---

Perform the steps in this topic to deploy PostgreSQL with Portworx on Kubernetes.

## Prerequisites

* **A running Portworx cluster**. Refer to the [Installation](/start-here-installation/#installation) page for details about how to install Portworx.

> * **Kubectl**. Refer to the [Install and Set Up kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) page for details about installing `kubectl`.

Reviewer: I usually specify which doc set I'm sending users to in these links to add clarity before a user clicks a link:
```suggestion
* **Kubectl**. Refer to the [Install and Set Up kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) page of the Kubernetes documentation for details about installing `kubectl`.
```

* **Stork**. Refer to the [Using Stork with Portworx](/portworx-install-with-kubernetes/storage-operations/stork/) page for details about installing Stork.

## Create a StorageClass

1. Create a file named `px-postgres-sc.yaml`, and copy in the following spec:

    ```text

>     #### Portworx StorageClass

Reviewer: Do we need these commented lines? I think they get in the way of the content more than they help describe it.

Author: Removed.

        type: RollingUpdate
      replicas: 1
      template:
        metadata:
          labels:
            app: postgres
        spec:
          schedulerName: stork
          affinity:
            nodeAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                nodeSelectorTerms:
                - matchExpressions:
                  - key: px/enabled
                    operator: NotIn
                    values:
                    - "false"
          containers:
          - name: postgres
            image: postgres:9.5
            imagePullPolicy: "IfNotPresent"
            ports:
            - containerPort: 5432
            env:
            - name: POSTGRES_USER
              value: pgbench
            - name: POSTGRES_PASSWORD
              value: superpostgres
            - name: PGBENCH_PASSWORD
              value: superpostgres
            - name: PGDATA
              value: /var/lib/postgresql/data/pgdata
            volumeMounts:
            - mountPath: /var/lib/postgresql/data
              name: postgredb
          volumes:
          - name: postgredb
            persistentVolumeClaim:
              claimName: postgres-data
      ```

>     There are several things to note about this example:

Reviewer: passive voice
```suggestion
    Note the following:
```

    * Specifies Stork as scheduler (`schedulerName: stork`)
    * Sets the following environment variables:
      * POSTGRES_USER (defines the superuser)
      * POSTGRES_PASSWORD (specifies the superuser password)
      * PGDATA (configures the location for the database files)
    * References the `postgres-data` PVC defined in the [Create a PVC](#create-a-pvc) section.

2. Apply the spec by entering the following command:

      ```text
      kubectl apply -f px-postgres-app.yaml
      ```

## Verify your PostgreSQL installation

> 1. Use the `kubectl get` command to list your storage classes:

Reviewer: I'm not a fan of `use` in this instance, I vote you stick with `enter`. I also would specify that it's the following command. Otherwise, it seems to indicate that just `kubectl get` would work.

```suggestion
1. Enter the following `kubectl get` command to list your storage classes:
```

      ```text
      kubectl get sc
      ```

      ```
      NAME             PROVISIONER                     AGE
      px-postgres-sc   kubernetes.io/portworx-volume   1h
      ```

    In the above example output, note that the provisioner is set to `kubernetes.io/portworx-volume`

2. Enter the `kubectl get pvc` command to verify that the PVC is bound to a volume:

      ```text
      kubectl get pvc
      ```

      ```
      NAME            STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS     AGE
      postgres-data   Bound     pvc-60e43292-06e3-11e8-96b4-022185d04910   1Gi        RWO            px-postgres-sc   1h
      ```

3. Use the `kubectl get pods` command to verify the status of the PostgreSQL pod:

      ```text
      kubectl get pod
      ```

      ```
      NAME                        READY     STATUS    RESTARTS   AGE
      postgres-86cb8587c4-l9r48   1/1       Running   0          1h
      ```

    Make a note of the name of the pod. You'll need it in the next step.

> 4. The following `kubectl exec` command opens a shell session into the `postgres-86cb8587c4-l9r48` pod:

Reviewer: This needs to be reworded as an instruction and made clear what action the user is performing and why.

```suggestion
4. Enter the following `kubectl exec` command to open a shell session into the `postgres-86cb8587c4-l9r48` pod:
```

      ```text
      kubectl exec -it postgres-86cb8587c4-l9r48 bash
      ```

      ```
      root@postgres-86cb8587c4-l9r48:/#
      ```

> 5. Start the PostgreSQL interactive shell. Use the`-U` flag to connect as user `pgbench`:

Reviewer: ```suggestion
5. Start the PostgreSQL interactive shell. Use the`-U` flag to connect as the `pgbench` user:
```
